crypto/src/main/java/net/corda/v5/crypto/DigitalSignature.java

```diff
@@ -0,0 +1,60 @@
+package net.corda.v5.crypto;
+
+import net.corda.v5.base.annotations.CordaSerializable;
+import net.corda.v5.base.types.OpaqueBytes;
+import org.jetbrains.annotations.NotNull;
+
+import java.security.PublicKey;
+import java.util.Map;
+
+/**
+ * A wrapper around a digital signature.
+ */
+@CordaSerializable
+public class DigitalSignature extends OpaqueBytes {
+    public DigitalSignature(@NotNull byte[] bytes) {
+        super(bytes);
+    }
+
+    /**
+     * A digital signature that identifies who the public key is owned by.
+     */
+    public static class WithKey extends DigitalSignature {
+
+        @NotNull
+        public final Map<String, String> getContext() {
+            return this.context;
+        }
+
+        /**
+         * Construct WithKey
+         *
+         * @param by      The public key of the corresponding private key used to sign the data (as if an instance
+         *                of the [CompositeKey] is passed to the sign operation it may contain keys which are not actually owned by
+         *                the member).
+         * @param bytes   The signature.
+         * @param context The context which was passed to the signing operation, note that this context is not signed over.
+         */
+        public WithKey(@NotNull PublicKey by, @NotNull byte[] bytes, @NotNull Map<String, String> context) {
+            super(bytes);
+            this.by = by;
+            this.context = context;
+        }
+
+        /**
+         * Public key which corresponding private key was used to sign the data (as if an instance
+         * of the [CompositeKey] is passed to the sign operation it may contain keys which are not actually owned by
+         * the member).
+         */
+        @NotNull
```

Comment:
Not sure that `@NotNull` makes much sense for `private` fields :thinking:.

Reply:
why not though? Still useful for compilers and linters

Reply:
comiitted locally